python/paddle/fluid/framework.py

```diff
@@ -2867,6 +2873,10 @@ def __init__(self):
         # appending gradients times
         self._appending_grad_times = 0
 
+        # when program is executed by exe.run, it will be set False
+        # When new op or var was append to program, it will be set True
+        self._add_new_elements = True
```

Comment:
I think adding flag here is not very elegant, but I have not thought of improving efficiency in a better way.